app/logic/serviceLearningCoursesData.py

```diff
@@ -90,3 +93,86 @@ def renewProposal(courseID, term):
                                 user=instructor.user)
 
     return newCourse
+
+def parseUploadedFile(filePath):
+    excelData = load_workbook(filename=filePath)
+    excelSheet = excelData.active
+    errorFlag = False
+    courseParticipantPreview= {}
+    errorList = []
+    previewTerm = ''
+    previewCourse = ''
+    studentValue= ''
+    cellRow = 1
+
+    for row in excelSheet.iter_rows():
+        cellVal = row[0].value
+        termReg = r"\b[a-zA-Z]{3,}\s\d{4}\b" # regular expression to check cells content
+        courseReg = r"\b[A-Z]{2,4}\s\d{3}\b"
+        bnumberReg = r"\b[B]\d{8}\b"
+
+        if regex.search(termReg, str(cellVal)):
+            hasTerm = Term.get_or_none(Term.description == cellVal)
+            previousTerm = list(Term.select().order_by(Term.termOrder))[-1].termOrder > Term.convertDescriptionToTermOrder(cellVal)
+            if hasTerm or previousTerm:
+                previewTerm = cellVal 
+            elif cellVal.split()[0] not in ["Summer", "Spring", "Fall", "May"]:
+                previewTerm = f"ERROR: {cellVal} is not valid."
+                errorFlag = True
+            else:
+                previewTerm = f"ERROR: The term {cellVal} does not exist."
+                errorFlag = True
+            courseParticipantPreview[previewTerm]= {}
+
+        elif regex.search(courseReg, str(cellVal)):
+            previewCourse = ''
+            hasCourse = Course.get_or_none(Course.courseAbbreviation == cellVal)
+            previewCourse = cellVal
+            if hasCourse and hasCourse.courseName:
+                previewCourse = hasCourse.courseName
+
+            courseParticipantPreview[previewTerm][previewCourse]=[]
+           
+        elif regex.search(bnumberReg, str(cellVal)):      
+            hasUser = User.get_or_none(User.bnumber == cellVal)
+            if hasUser:
+                studentValue =f"{hasUser.firstName} {hasUser.lastName}"
+            else:
+                previewStudent = row[1].value                
```

Comment:
I don't think we need a variable here, since `row[1].value` is short and is just used once in the next line.